ui/src/App.tsx

```diff
@@ -189,27 +191,80 @@ const App = () => {
     setEvents(expandedEvents);
   };
 
-  const handleDeleteEntry = async (e: MouseEvent<HTMLButtonElement>) => {
-    e.preventDefault();
-    deleteEntry(displayedEventData._id!)
+  const setEventsWithExtraFields = (entries: any) => {
+    // We need to manually set the entryStart field
+    // so we can read it off the object later
+    // See https://fullcalendar.io/docs/event-parsing
+    // Can't use the 'start' field because it gets truncated
+    // for all day events when coming back from FullCalendar
+    const expandedEntries = entries.map((entry: any) => {
+      return {
+        ...entry,
+        entryStart: entry.start,
+      };
+    });
+
+    setEvents(expandedEntries);
+  };
+
+  const handleDeleteSeries = () => {
+    deleteEntry(displayedEventData._id!, displayedEventData.startTimeUtc, true)
       .then(() => {
         getEntries(rangeStart, rangeEnd).then((entries) => {
           setEventsWithStart(entries);
+          setShowDeletionSelectionScreen(false);
+          setShowOverlay(false);
+        });
+      })
+      .catch(() => {
+        setShowOverlay(false);
+        setShowDeletionSelectionScreen(false);
+        flashApiErrorMessage();
+      });
+  };
+
+  const handleDeleteRecurringInstance = () => {
+    deleteEntry(displayedEventData._id!, displayedEventData.startTimeUtc, false)
+      .then(() => {
+        getEntries(rangeStart, rangeEnd).then((entries) => {
+          setEventsWithExtraFields(entries);
+          setShowDeletionSelectionScreen(false);
           setShowOverlay(false);
         });
       })
       .catch(() => {
         setShowOverlay(false);
+        setShowDeletionSelectionScreen(false);
         flashApiErrorMessage();
       });
   };
 
+  const handleDeleteEntry = async (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (displayedEventData.recurring) {
+      setShowDeletionSelectionScreen(true);
+    } else {
+      deleteEntry(displayedEventData._id!)
+        .then(() => {
+          getEntries(rangeStart, rangeEnd).then((entries) => {
+            setEventsWithExtraFields(entries);
```

Comment:
We renamed this to something similar to 'SetEventsWithStartTime' in a different PR and can update the name here too.